GraphvizBlock.hs

```diff
@@ -1,42 +1,58 @@
-                           _ -> CodeBlock (id, classes, keyvals) err
-          Format "latex" ->
-              do
-                (ec, out, err) <- readProcessWithExitCode
-                                  "dot2tex" ["--figonly", "--progoptions=-K" ++ layout] content
-                return $ if length err == 0
-                         then RawBlock (Format "latex") out
-                         else CodeBlock (id, classes, keyvals) err
-          _ ->
-              do
-                (ec, out, err) <- readProcessWithExitCode
-                                  "dot" ["-Tpng", "-K" ++ layout, "-o" ++ filename] content
-                return $ case ec of
-                           ExitSuccess -> Para [Image (id, [], []) [Str content] (filename, id)]
-                           _ -> CodeBlock (id, classes, keyvals) err
-              where
-                uniq = ((showDigest . sha256 . fromString) (layout ++ "/" ++ content))
-                filename = "graphviz-" ++ uniq ++ ".png"
-    where
-      layout =
-          case lookup "layout" keyvals of
-            Just x -> x
-            _ -> "dot"
+graphvizBlock (Just format) (CodeBlock (ident, classes, keyvals) content)
+    | "graphviz" `elem` classes = case format of
+        Format "html" -> do
+            (ec, out, err) <-
+                readProcessWithExitCode "dot" ["-Tsvg", "-K" <> layout]
+                    $ T.unpack content
+            return $ case ec of
+                ExitSuccess -> RawBlock
+                    (Format "html")
+                    (  "<div ident=\""
+                    <> ident
+                    <> "\" class=\"graphviz\">"
+                    <> T.pack out
+                    <> "</div>"
+                    )
+                _ -> CodeBlock (ident, classes, keyvals) $ T.pack err
+        Format "latex" -> do
+            (_, out, err) <-
+                readProcessWithExitCode
+                        "dot2tex"
+                        ["--figonly", "--progoptions=-K" ++ layout]
+                    $ T.unpack content
+            return $ if null err
+                then RawBlock (Format "latex") $ T.pack out
+                else CodeBlock (ident, classes, keyvals) $ T.pack err
+        _ -> do
+            (ec, _, err) <-
+                readProcessWithExitCode
+                        "dot"
+                        ["-Tpng", "-K" ++ layout, "-o" ++ filename]
+                    $ T.unpack content
+            return $ case ec of
+                ExitSuccess -> Para
+                    [Image (ident, [], []) [Str content] (T.pack filename, ident)]
+                _ -> CodeBlock (ident, classes, keyvals) $ T.pack err
+          where
+            uniq = (showDigest . sha256 . fromString)
+                (layout <> "/" <> T.unpack content)
+            filename = "graphviz-" <> uniq <> ".png"
+  where
+    layout = T.unpack $ case lookup "layout" keyvals of
+        Just x -> x
+        _      -> "dot"
```

Comment:
Most of changes seem style formatting. Can you keep the original code as much as possible to clarify the essential changes?

Reply:
If current code does not pass lints, can you make another PR for fixing styles?